front/pages/api/v1/w/[wId]/apps/[aId]/runs/index.ts

```diff
@@ -264,92 +186,194 @@ async function handler(
         });
       }
 
-      await Run.create({
-        dustRunId: runRes.value.run.run_id,
-        appId: app.id,
-        runType: "deploy",
-        workspaceId: keyRes.value.workspaceId,
-      });
-
-      let run: RunType = runRes.value.run;
-      run.specification_hash = run.app_hash;
-      delete run.app_hash;
-
-      // If `blocking` is set, poll for run completion.
-      if (req.body.blocking) {
-        const runId = run.run_id;
-        try {
-          await poll({
-            fn: async () => {
-              const run = await coreAPI.getRunStatus({
-                projectId: app.dustAPIProjectId,
-                runId,
+      if (req.body.stream) {
+        // Start SSE stream.
+        res.writeHead(200, {
+          "Content-Type": "text/event-stream",
+          "Cache-Control": "no-cache",
+          Connection: "keep-alive",
+        });
+      } else if (!req.body.blocking) {
+        // Non blocking, return a run object as soon as we get the runId.
+        void (async () => {
```

Comment:
Why void here vs await?

Reply:
We can't await here as the `await runRes.value.dustRunId` will be resolved only during stream execution (when receiving the first event), which is done later in the code. On the other hand we don't want to wait for the full event stream to be processed to reply, as we are not blocking - so we can't put this code after the event loop handling.
Another solution would be to not rely on `value.dustRunId` and listen to `run_status` events in the event loop instead, sends a response the first time we receive a `run_status` with an id  - but imho this makes the code much more confusing.